### PR TITLE
Initial addition of Neptune

### PR DESCRIPTION
**Pull request by ghost**
- Update eslint rules based on inappropriate errors reported
- Simplify CDK deployments to use a single CloudFormation stack per stage
- Add AWS resources to support the migration to Neptune:
  - A Neptune database cluster
  - A bulk loader bucket. This is used for bulk loading data into the Neptune cluster.
  - An ECS task for handling the bulk loader and re-indexing operations. The Docker image to be used is defined in `neptune-bulk-loader/docker`
  - A `TaskLauncher` lambda function defined in `neptune-bulk-loader/task-launcher.ts`. When a metadata file is written into the `bulk-loader-trigger/` path of the bulk loader bucket, this lambda function is triggered. It is responsible for starting the ECS task that will perform the bulk load and re-indexing.

### CODE REVIEW

File: bin/app.ts [hunk 1]
```diff
@@ -4,30 +4,41 @@ import 'source-map-support/register';
 import {
   App,
   Environment,
+  Tags,
 } from '@aws-cdk/core';
-import Obp from '../cdk/obp';
+import ObpStack from '../cdk/obp-stack';
 
-const env: Environment = {
+if (process.env['CDK_DEFAULT_ACCOUNT'] === undefined) {
```
Comment (paulpilone) [inline bin/app.ts]: Check for empty string.

File: cdk/neptune.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    this.neptuneCluster = new neptune.DatabaseCluster(this, 'Database', {
+      dbClusterName: stackName,
+      instanceType: neptune.InstanceType.T3_MEDIUM,
+      vpc,
+      vpcSubnets: { subnetType: ec2.SubnetType.PUBLIC },
+      removalPolicy,
+      associatedRoles: [neptuneRole],
+    });
+
+    for (const other of allowFrom) {
```
Comment (paulpilone) [inline cdk/neptune.ts:56]: Make sure `/sparql` API endpoint can connect to the cluster.

**Review by paulpilone — CHANGES_REQUESTED** [review]

File: cdk/neptune.ts (hunk 2 above)
Comment (ghost) [reply cdk/neptune.ts:56]: Fixed in 050061e2b206ed3ade309124b2cea05553496ddd

File: bin/app.ts (hunk 1 above)
Comment (ghost) [reply bin/app.ts]: Fixed in 3cca8c3e8fb4c88fba1b40f32347ae164e13ac78

File: cdk/ingest/lambdas.ts [hunk 3]
```diff
@@ -127,7 +124,7 @@ export default class IngestLambdas extends Construct {
     snsTopics.availableDocument.grantPublish(this.scheduler);
 
     this.indexRectifier = new Function(this, 'IndexRectifier', {
-      functionName: `${stage}-obp-cdk-index-rectifier`,
+      functionName: `${stackName}-index-rectifier`,
       handler: 'handler.handler',
```
Comment (paulpilone) [inline cdk/ingest/lambdas.ts]: Does this need to be changed to `lambda`?

**Review by paulpilone — CHANGES_REQUESTED** [review]

File: cdk/ingest/lambdas.ts (hunk 3 above)
Comment (ghost) [reply cdk/ingest/lambdas.ts]: Fixed in c892d65

**Pull request merged**